File: min_word_dist1.py
# // Time Complexity :O(n)
# // Space Complexity :O(1)
# // Did this code successfully run on Leetcode :yes
# // Any problem you faced while coding this :no

class Solution:
    def calc(self,worddict,word1,word2):
        l1=-1
        l2=-1
        minn=999999999
        for i in range(len(worddict)):
            if worddict[i]==word1:
                l1=i
            elif worddict[i]==word2:
                l2=i
            if(l1!=-1 and l2!=-1):
                minn=min(minn,abs(l1-l2))
        return minn
obj=Solution()
# print(obj.calc(["practice","makes","perfect","coding","makes"],"coding","practice"))
print(obj.calc(["practice","makes","perfect","coding","makes"],"coding","makes"))
                
            
        

File: MinWordDist1.java
// Time Complexity : O(n * l) where n is the no. of words and l is the length of word
// Space Complexity : O(1)
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this : No


// Your code here along with comments explaining your approach
// Take two pointers 1 each for word1 and word2
// If you see word 1 assign index to it, do same for word2
// Check dif between these each time and keep track of min 
// Return the min value
class Solution {
    public int shortestDistance(String[] wordsDict, String word1, String word2) {
        int p1 = -1;
        int p2 = -1;
        int min = Integer.MAX_VALUE;
        for(int i = 0; i < wordsDict.length; i++){
            String curword = wordsDict[i];
            if(curword.equals(word1))
                p1 = i;
            else if(curword.equals(word2))
                p2 = i;
            if(p1 != -1 && p2 != -1){
                int dif = Math.abs(p1 - p2);
                min = Math.min(min, dif);
            }
        }
        return min;
    }
}